packages/react-router/modules/Redirect.js

```diff
@@ -43,7 +43,11 @@ function Redirect({ computedMatch, to, push = false }) {
               method(location);
             }}
             onUpdate={(self, prevProps) => {
-              if (!locationsAreEqual(prevProps.to, location)) {
+              if (typeof prevProps.to === "string") {
+                if (prevProps.to !== location.pathname) {
```

Comment:
What happens if the `to` string includes `hash` or `search` components? I think that you would need to use `createLocation` to create a location from `prevProps.to` and the compare it to the `location` using `locationsAreEqual`.
```js
const prevLocation = typeof prevProps.to === "string"
  ? createLocation(prevProps.to)
  : prevProps.to;
if (!locationsAreEqual(prevLocation, location) {
  method(location);
}
```

Reply:
Done.